pylint/checkers/variables.py

```diff
@@ -858,10 +859,8 @@ def leave_functiondef(self, node):
         if is_method and node.is_abstract():
             return
 
-        global_names = _flattened_scope_names(node.nodes_of_class(astroid.Global))
```

Comment:
Moving this one inside the loop will make it be called every time we have an unconsumed value, is that right?

Reply:
Ah, I see what you mean. In practice, that part of the loop might never get reached, in which case the nodes will never need to be calculated at all. I'll leave it alone for now though.